### DISCUSSION TITLE
question: how to use long for custom scalar

### DISCUSSION THREAD

**Discussion by tenstad** [ExpediaGroup/graphql-kotlin#1725] [Q&A]
> Custom scalar support - defaults to String but can be configured to deserialize to specific types

Is it possible to have the custom `Long` scalar be a `Long` instead of a `String` as the underlying type?  
It is now quite strange to assert that the long values returned by the test-client is in fact strings.

Using this custom scalar:
```kotlin
    override fun willGenerateGraphQLType(type: KType): GraphQLType? = when (type.classifier) {
        Long::class -> ExtendedScalars.GraphQLLong
```
results in:
```kotlin
/**
 * Long type
 */
public typealias Long = String
```

**Comment by dariuszkuc** [comment]
Hello,

You need to configure your client with an appropriate converter -> [docs](https://opensource.expediagroup.com/graphql-kotlin/docs/client/client-features#custom-scalar-support).

Thanks,
Derek
